Source-build encounters circular PublishBuilder dependency in WasmAppBuilder

This issue was discovered while updating source-build to the latest 6.0 commit shas - https://github.com/dotnet/installer/pull/11710/files

/repos/tarball1/src/runtime.c300b096419523024f2b807ec9db3c2d91df0298/artifacts/source-build/self/src/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj(33,5): error MSB4006: There is a circular dependency in the target dependency graph involving target "PublishBuilder".

Binlog - [runtime.circularbuild.sourcebuild.zip](https://github.com/dotnet/runtime/files/7130269/runtime.circularbuild.sourcebuild.zip)

This is what the call graph looks like:
MSBuild WasmAppBuilder net6.0;net472
	WasmAppBuilder net6.0 -> Build
		WasmAppBuilder net472 -> publish
			MSBuild
			    Parameters
			        Projects = WasmAppBuilder.csproj
			        Targets = Publish
			        Properties = TargetFramework=net472
			    Global Properties:
			        TargetFramework = net472

This appears to be caused by this target in WasmAppBuilder.csproj - https://github.com/dotnet/runtime/blob/release/6.0/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj#L26

```
  <Target Name="PublishBuilder"
          AfterTargets="Build">

    <!-- needed for publishing with multi-targeting. We are publishing essentially to get the SR.MetadataLoadContext.dll :/ -->
    <ItemGroup>
      <_PublishFramework Include="$(TargetFrameworks)" />
    </ItemGroup>
    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Publish" Properties="TargetFramework=%(_PublishFramework.Identity)" />
  </Target>
```

--- a/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
+++ b/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
@@ -24,13 +24,14 @@
   </ItemGroup>
 
   <Target Name="PublishBuilder"
-          AfterTargets="Build">
+          AfterTargets="Build"
+          Condition="'$(_RunningForPublishBuilder)' != 'true'">
 
     <!-- needed for publishing with multi-targeting. We are publishing essentially to get the SR.MetadataLoadContext.dll :/ -->
     <ItemGroup>
       <_PublishFramework Include="$(TargetFrameworks)" />
     </ItemGroup>
-    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Publish" Properties="TargetFramework=%(_PublishFramework.Identity)" />
+    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Publish" Properties="TargetFramework=%(_PublishFramework.Identity);_RunningForPublishBuilder=true" />
   </Target>
 
   <Target Name="GetFilesToPackage" Returns="@(FilesToPackage)">

